packages/core/src/dropdown/Dropdown.css

```diff
@@ -49,52 +67,102 @@
 
 .saltDropdown-primary {
   --dropdown-background: var(--salt-editable-primary-background);
+  --dropdown-background-active: var(--salt-editable-primary-background-active);
+  --dropdown-background-hover: var(--salt-editable-primary-background-hover);
+  --dropdown-background-disabled: var(--salt-editable-primary-background-disabled);
+  --dropdown-background-readonly: var(--salt-editable-primary-background-readonly);
 }
 
 .saltDropdown-secondary {
   --dropdown-background: var(--salt-editable-secondary-background);
+  --dropdown-background-active: var(--salt-editable-secondary-background-active);
+  --dropdown-background-hover: var(--salt-editable-secondary-background-active);
+  --dropdown-background-disabled: var(--salt-editable-secondary-background-disabled);
+  --dropdown-background-readonly: var(--salt-editable-secondary-background-readonly);
 }
 
-.saltDropdown:disabled {
-  --dropdown-background: var(--salt-editable-primary-background-disabled);
+.saltDropdown:disabled,
+.saltDropdown:disabled:active,
```

Comment:
Do we really need all of these selectors?

Reply:
Otherwise for validation status disabled case, the disabled css is not applied for diff states. tried to keep same implementation as that of input.